# Задачи по Эконометрике: Интерпретация коэффициентов
Савелий Захаров
# Output Equation

## Output Equation (Модель 1)

На датасете `Labour` рассмотрим региессию **log(output) на log(capital), log(labour)**

Спецификация: $\log(output)=\beta_0+\beta_1\log(capital)+\beta_2log(labour)+u$

In [33]:
import numpy as np # linear algebra 
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
# stats models: regression fitting via formulas
import statsmodels.formula.api as smf
# stats models: regression fitting via matrices of regression design
import statsmodels.api as sm

In [34]:
df = pd.read_csv('/Users/saveliyzakharov/Downloads/Econometrica-master-2/python-notebooks/data-csv/Labour.csv')
df.head()

capital  labour     output       wage
0   2.606563     184   9.250759  43.080307
1   1.323237      91   3.664310  27.780016
2  22.093692     426  28.781516  44.467748
3  10.737851      72   4.124642  39.734710
4   1.161365      46   2.890150  34.650709

### Способ оценивания через спецификацию

In [35]:
# специфицируем модель через формулу
output_eq1 = smf.ols(formula='np.log(output)~np.log(capital)+np.log(labour)', data=df).fit()
# Коэфициенты модели с округление
output_eq1.params.round(3)

Intercept         -1.711
np.log(capital)    0.208
np.log(labour)     0.715
dtype: float64

### Интерпретация модели
1) При увеличении капитала на 1%, выпуск увеличивается на 0.208%
2) При увеличении количества рабочих на 1%, выпуска возрастает на 0.715% при прочих равных условиях

## Output Equation (Модель 2)

На датасете `Labour` рассмотрим региессию **log(output) на log(capital), log(labour), log(wage)**

Спецификация: $\log(output)=\beta_0+\beta_1\log(capital)+\beta_2log(labour)+\beta_3log(wage)+u$

### Способ оценивания через спецификацию

In [38]:
# специфицируем модель через формулу
output_eq1 = smf.ols(formula='np.log(output)~np.log(capital)+np.log(labour)+np.log(wage)', data=df).fit()
# Коэфициенты модели с округление
output_eq1.params.round(3)

Intercept         -5.007
np.log(capital)    0.149
np.log(labour)     0.720
np.log(wage)       0.921
dtype: float64

### Интерпретация модели
1) При увеличении количества рабочих на 1%, выпуск возрастает на 0.72%
2) При увеличении капитала на 1%, выпуск увеличивается на 0.149%, при прочих равных